common/djangoapps/enrollment/views.py

```diff
@@ -345,12 +363,23 @@ def post(self, request):
             )
 
         try:
-            response = api.add_enrollment(user, unicode(course_id))
+            # Check to see if the enrollment mode is something other than honor. If so, check to see if the user
+            # is enrolled, and if so, perform an upgrade instead of creating a new enrollment.
+            enrollment = {}
+            http_success_status = status.HTTP_200_OK
+            if mode is not 'honor':
```

Comment:
Yep I definitely need some logic to check if the user is already enrolled before attempting to enroll again. I also need to check to see if the enrollment exists, but is inactive. I think I'm going to take a minute to clean this up so it's not so cluttered.